### ISSUE TITLE
'Type of bollard?'-Quest for first responders 

### ISSUE THREAD

**Issue by RubenKelevra** [streetcomplete/StreetComplete#2128]
This information is quite useful for routing first responders vehicles, like police, ambulances and fire brigades to a destination. 

### General
Affected tag(s) to be modified/added: [bollard=*](https://wiki.openstreetmap.org/wiki/Key:bollard)
Question asked: **What type of bollard is this?**

### Checklist
Checklist for quest suggestions (see guidelines):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary 
- [x] 💤 Not an overwhelming percentage of elements have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of elements (Worth the effort)

**Comment by RubenKelevra** [comment]
Cycle barriers can also often be removed, to give first responders access. It might make sense to have a similar or the same quest for both.

**Comment by westnordost** [comment]
Hmm, not sure how valuable this information is. After all, if you do **not** have the key, every bollard will be a `fixed` one for you. Who has the key? The proprietor, but he doesn't need a map to use this information. The emergency services, but they also do not need that information because they will know which locks fit on the keys they have.
I assume that for these bollards, there are standard keys. I'd also assume that for emergency vehicle access, at least in Germany, practically every bollard where if the bollard is removed, a vehicle can fit through, is a foldable one. I am pretty sure that is required by law.

To summarize, I can't think of any use case where this information would be useful.

**Comment by peternewman** [comment]
> I assume that for these bollards, there are standard keys.

Yes they are in London/UK:
https://www.locksonline.co.uk/community/what-are-fb-padlocks-locks-and-keys.html

I'd also imagine unless you want access directly behind the bollard, driving round via another route may well be faster than stopping and faffing with a key, I've never seen one used that I remember.

**Comment by RubenKelevra** [comment]
Well, the keys are standardized - so every first responder vehicle has a key. It takes stopping and like half a minute of work to remove it. The key means, it is removable or foldable by the first responder key.

Additionally there are raising lowering bollards which are operated by transponders in busses to avoid that cars drive in streets where only busses are allowed to drive in. 

In my area there a a lot of stone/concrete bollards, as well as plastic ones which can be driven over, but look like old metal ones:

![altstadtpoller_90_mm](https://user-images.githubusercontent.com/614929/95059814-b2200b80-06f9-11eb-803c-07a2347f97c3.jpg)

The main reason to collect this information is that many streets are only accessible in one direction, the other side is usually fixed. 

So you need to have the information right when you are approaching a scene, to avoid having to turn a large firetruck in a narrow street and drive around to get to the street in the different direction. 

Having this informations collected would increase the usefulness of OpenStreetMap quite a lot for first responders!

**Comment by westnordost** [comment]
Do you happen to have a photo of the complete scene for the metal ones? Cause I would expect that there is at least one amongst those which is removable or it is not relevant for emergency services, for example if they are just use to segregate the area where cars may drive from the sidewalk.

**Comment by RubenKelevra** [comment]
> Do you happen to have a photo of the complete scene for the metal one? Cause I would expect that there is at least one amongst those which is removable or it is not relevant for emergency services, for example if they are just use to segregate the area where cars may drive from the sidewalk.

Well this one is not made out of metal, just supposed to look like one. It's made out of plastic and an emergency vehicle can drive over it. 

Point was more about that it's not obvious where you can drive through in such cases, so the data needs to be collected or emergency vehicles need to scout out on which side the flexible post is while responding. 

A typical scenario is this one:

This cycle barrier can be removed for emergency access:

https://www.openstreetmap.org/node/1458626596

While just some hundred meters away is one which cannot be removed, since it's mounted permanently in concrete:

https://www.openstreetmap.org/node/371218844

In the center is another street where access for vehicles is blocked by large boulders on one side.

The difference is the age of construction - one side was made in the 90th with fixed cycle barriers while the other side was a recent addition.

Those kinds of local details do often cause first responders to loose valuable time to reach their destinations. They currently use mainly Google Maps and hope for the best that they can drive into a footway from this direction. 

The more data like this is added to the database the easier it gets to create routing software which can properly route them to their destination.

**Comment by westnordost** [comment]
Emergency services use google maps?

**Comment by RubenKelevra** [comment]
> Emergency services use google maps?

Google Maps offer very good traffic predictions. If the route is long and they expect traffic they tend to use it or if they have to respond to an area the driver isn't totally familiar with. 

This is obviously only the case if the emergency vehicle is not equipped with a commercial navigation system, specially designed for first responders. 

A friend works in large German city on ambulances and they have no navigation equipment on board. They solely use Google Maps to find their route and their destination.

**Comment by RubenKelevra** [comment]
I've looked at the data currently in the database and don't see any indication that bollards are in general removable in a certain country:

![Screenshot_20201006-211959](https://user-images.githubusercontent.com/614929/95251354-b3e1f000-081b-11eb-8a23-e5af8b59d8b8.png)


I think the default assumption is, that bollards are fixed, and if the mapper cared to check and it was removable they added the second tag for this. That's why there are not many bollards mapped with bollard=fixed.

But as I said, that's a pretty crucial information on first responder routing, especially for smaller vehicles like police cars and ambulances on pedestrian, hiking, cycling tracks - where sometimes it's only accessible from one side or not at all.

Checking out both sides in a case where no access is possible for example would waste a lot of time the first responders could have already used to approach the accident scene by foot.

Currently only 3.94% of bollards[1] are explicitly tagged that they are removable. So there's a lot of potential to add explicit informations. 

[1] https://taginfo.openstreetmap.org/tags/barrier=bollard#combinations

**Comment by westnordost** [comment]
Ok, let's add it then. Maybe free-floating bollards (those that are not nodes on a `higwhay`) should be excluded because it may be difficult to understand in the question which one is meant if there are several close to each other.

By the way, I was looking at a bollard today and I could honestly not find out if it was foldable or removable or something else. I saw a lock on it, but it didn't look like it could be folded.

As usual, finding good and free images for the bollards will be the most effort.

**Comment by RubenKelevra** [comment]
> Ok, let's add it then. Maybe free-floating bollards (those that are not nodes on a `higwhay`) should be excluded because it may be difficult to understand in the question which one is meant if there are several close to each other.

True, could be difficult on multiple bollards in the same place... are they part of the map-data? Maybe we can render them as points, similar to mapnik - this should make it easier to understand where which one is located.

> By the way, I was looking at a bollard today and I could honestly not find out if it was foldable or removable or something else. I saw a lock on it, but it didn't look like it could be folded.

Foldable is easy: There's a clearly visible hinge at the bottom. It's usually the square type which is foldable, while the round ones are removable.

When we do the quest, we should really do both: cycle barriers and bollards together. They are so similar that we only need to replace the picture and the tag where we put the information - the request to the server could fetch both. If that makes sense for the codebase and the quest could just contain both in one icon.

> As usual, finding good and free images for the bollards will be the most effort.

I can do a walk around and take some pictures. Should be possible to find all common types here.

**Comment by RubenKelevra** [comment]
by @RubenKelevra
> by @westnordost
> > As usual, finding good and free images for the bollards will be the most effort.
> 
> I can do a walk around and take some pictures. Should be possible to find all common types here.

I took some pictures of different types of barriers but didn't come across any cycle barriers. Maybe next time :)

https://ipfs.io/ipfs/bafybeiha3yer5te2tv2myhafrfrhpss5eummdtjwub4eaznra7yiw5u47u

**Comment by westnordost** [comment]
Well done! 

Maybe you want to propose some pictures by posting their thumbnails here?, the pictures on ipfs load kind of slow. I have looked at a few, I'd suggest to take those that look "clean", so those that don't have shit on them (graffity, stickers, ...) and also don't have much stuff in the background that distract (leaves, ...). The pictures chosen should of course look generic.

**Comment by RubenKelevra** [comment]
@westnordost will do another trip in the next days and try to get pictures for any type used in the area.

The pictures on IPFS load alright for me, they are just pretty large. 

I will select the best ones and share them with a more reasonable size - together with the full size if there's a need to do any additional work on them in the future.

**Comment by RubenKelevra** [comment]
I did two walks but couldn't find those bicycle gates in the vicinity. Those which were there some years ago got replaced by bollards to ease access for bikes.

But I found a good picture which is public domain for them:

https://commons.wikimedia.org/wiki/File:211-Sambatrasse-22km.jpg

**Comment by RubenKelevra** [comment]
Here are the pictures which should be necessary:

Bollards removable with key:

![IMG_20201012_164210_Bokeh_processed](https://user-images.githubusercontent.com/614929/96745941-efa5b980-13c6-11eb-953a-d2a74f75d086.jpg)

Keyhole detail:

![IMG_20201012_164218_processed](https://user-images.githubusercontent.com/614929/96747109-2defa880-13c8-11eb-96a5-119e1fb0ec37.jpg)

Bollard removable without key:

![IMG_20201012_161230_processed](https://user-images.githubusercontent.com/614929/96747939-35638180-13c9-11eb-961c-6fee8e98067c.jpg)

Bollard without key detail:

![IMG_20201012_161235_processed](https://user-images.githubusercontent.com/614929/96748272-99864580-13c9-11eb-9e56-2413842ecb7c.jpg)

Bicycle gate (Wikimedia image):

![211-Sambatrasse-22km_preprocessed](https://user-images.githubusercontent.com/614929/96749697-7066b480-13cb-11eb-93fd-ea6771aff85d.jpg)

**Comment by RubenKelevra** [comment]
I found a flexible bollard without any stickers :)

![MVIMG_20201023_101713_preprocessed](https://user-images.githubusercontent.com/614929/97271508-ee461800-1830-11eb-967b-69c69ee13b1f.jpg)
License: CC0

**Comment by RubenKelevra** [comment]
I just wanted to add, that the German first responders already use OpenStreetMap as their default map. It's currently the standard layer, but maybe someone will add details like this to a specialized layer.

![20210521_044142](https://user-images.githubusercontent.com/614929/119074192-df060200-b9ee-11eb-93e0-3e2a74ec25a4.jpg)

**Comment by FloEdelmann** [comment]
So, to recap:

Ask for all nodes on a `highway` with `barrier=bollard and !bollard`: **"What type of bollard is this?"**

* rising → `bollard=rising`
  <img src="https://wiki.openstreetmap.org/w/images/d/dc/Rising_Bollard.jpg" width="200"> (<a href="https://wiki.openstreetmap.org/wiki/File:Rising_Bollard.jpg">image source</a>)
* removable with key → `bollard=removable`
  <img src="https://user-images.githubusercontent.com/614929/96747109-2defa880-13c8-11eb-96a5-119e1fb0ec37.jpg" width="200"> (<a href="https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-713700580">image source</a>)
* removable without key → also `bollard=removable`
  <img src="https://user-images.githubusercontent.com/614929/96748272-99864580-13c9-11eb-9e56-2413842ecb7c.jpg" width="200"> (<a href="https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-713700580">image source</a>)
* foldable → `bollard=foldable`
  <img src="https://wiki.openstreetmap.org/w/images/8/8d/Bollard_foldable.jpg" width="200"> (<a href="https://wiki.openstreetmap.org/wiki/File:Bollard_foldable.jpg">image source</a>)
* flexible → `bollard=flexible`
  <img src="https://user-images.githubusercontent.com/614929/97271508-ee461800-1830-11eb-967b-69c69ee13b1f.jpg" width="200"> (<a href="https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-717054203">image source</a>)
* fixed → `bollard=fixed`
  <img src="https://wiki.openstreetmap.org/w/images/4/4b/Bollard_in_residential_area.png" width="200"> (<a href="https://wiki.openstreetmap.org/wiki/File:Bollard_in_residential_area.png">image source</a>)

Did I get this right?

---

> When we do the quest, we should really do both: cycle barriers and bollards together. They are so similar that we only need to replace the picture and the tag where we put the information

I don't get what you want to say with this. The `bollard=*` tag does not make sense on nodes with `barrier=cycle_barrier`, does it? And there is no corresponding `cycle_barrier=*` tag with similar values.

**Comment by westnordost** [comment]
A note on the photos - they are all bollards... I guess it would make sense to zoom in a bit on the features of the individual bollards that matter, f.e. the foldable part of that foldable bollard etc.

**Comment by FloEdelmann** [comment]
Suggestion for a quest icon:

![bollard](https://user-images.githubusercontent.com/202916/119276114-51f2c100-bc19-11eb-8b24-c7745d952220.png)

[bollard.svg](https://github.com/streetcomplete/StreetComplete/files/6528902/bollard.zip)

**Comment by westnordost** [comment]
Maybe like the "prohibited for pedestrian" icon but for cars? Because, basically that is what they are for, aren't they?

**Comment by FloEdelmann** [comment]
> Maybe like the "prohibited for pedestrian" icon but for cars?

Like this?

![no_cars](https://user-images.githubusercontent.com/202916/119401134-60a1ac80-bcdb-11eb-8f71-d32b15343bb4.png) [no_cars.svg](https://github.com/streetcomplete/StreetComplete/files/6534700/no_cars.zip)

**Comment by westnordost** [comment]
Yes, like this, very cool!

Though, why does the car look so happy if it is not allowed in? Could maybe also be even more ghostbusters-style (car overlapping with one of side of the red circle)

**Comment by FloEdelmann** [comment]
Next iteration:

![no_cars](https://user-images.githubusercontent.com/202916/119452239-67114200-bd36-11eb-8fb0-c13496b63ea9.png) [no_cars.svg](https://github.com/streetcomplete/StreetComplete/files/6537116/no_cars.zip)

**Comment by westnordost** [comment]
Yay!

Am 25. Mai 2021 08:52:11 MESZ schrieb Flo Edelmann ***@***.***>:
>Next iteration:
>
>![no_cars](https://user-images.githubusercontent.com/202916/119452239-67114200-bd36-11eb-8fb0-c13496b63ea9.png)
>[no_cars.svg](https://github.com/streetcomplete/StreetComplete/files/6537116/no_cars.zip)
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-847595004

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by FloEdelmann** [comment]
For the rising bollard, this is probably a better image (when cropped correctly):

<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1e/Rising_bollards_in_Moscow.jpg/640px-Rising_bollards_in_Moscow.jpg" width="200"> (<a href="https://commons.wikimedia.org/wiki/File:Rising_bollards_in_Moscow.jpg">image source</a>)

**Comment by RubenKelevra** [comment]
> So, to recap:
> 
> Ask for all nodes on a `highway` with `barrier=bollard and !bollard`: **"What type of bollard is this?"**
> 
> * rising → `bollard=rising`
>   <img src="https://wiki.openstreetmap.org/w/images/d/dc/Rising_Bollard.jpg" width="200"> (<a href="https://wiki.openstreetmap.org/wiki/File:Rising_Bollard.jpg">image source</a>) or <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/1e/Rising_bollards_in_Moscow.jpg/640px-Rising_bollards_in_Moscow.jpg" width="200"> (<a href="https://commons.wikimedia.org/wiki/File:Rising_bollards_in_Moscow.jpg">image source</a>)
> * removable with key → `bollard=removable`
>   <img src="https://user-images.githubusercontent.com/614929/96747109-2defa880-13c8-11eb-96a5-119e1fb0ec37.jpg" width="200"> (<a href="https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-713700580">image source</a>)
> * removable without key → also `bollard=removable`
>   <img src="https://user-images.githubusercontent.com/614929/96748272-99864580-13c9-11eb-9e56-2413842ecb7c.jpg" width="200"> (<a href="https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-713700580">image source</a>)
> * foldable → `bollard=foldable`
>   <img src="https://wiki.openstreetmap.org/w/images/8/8d/Bollard_foldable.jpg" width="200"> (<a href="https://wiki.openstreetmap.org/wiki/File:Bollard_foldable.jpg">image source</a>)
> * flexible → `bollard=flexible`
>   <img src="https://user-images.githubusercontent.com/614929/97271508-ee461800-1830-11eb-967b-69c69ee13b1f.jpg" width="200"> (<a href="https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-717054203">image source</a>)
> * fixed → `bollard=fixed`
>   <img src="https://wiki.openstreetmap.org/w/images/4/4b/Bollard_in_residential_area.png" width="200"> (<a href="https://wiki.openstreetmap.org/wiki/File:Bollard_in_residential_area.png">image source</a>)
> 
> Did I get this right?


Yes 


> 
> ---
> 
> > When we do the quest, we should really do both: cycle barriers and bollards together. They are so similar that we only need to replace the picture and the tag where we put the information
> 
> @RubenKelevra I don't get what you want to say with this. The `bollard=*` tag does not make sense on nodes with `barrier=cycle_barrier`, does it? And there is no corresponding `cycle_barrier=*` tag with similar values.


Well, using the value of a key as subkey is pretty common. I doubt someone would complain if we 'invent' `cycle_barrier=removeable` as an analog to `bollard=removeable`. 

This info is just very uncommon to be added right now, but this would follow established procedures and is analog to a very similar tag. 

That's what I meant with that :)

**Comment by FloEdelmann** [comment]
Ah, that makes it clear, thanks. I'm not sure if it's okay for StreetComplete to start tagging that new tag though. There are already some (few) uses of the tag: https://taginfo.openstreetmap.org/keys/cycle_barrier#values

@westnordost What's your opinion on that?

**Comment by RubenKelevra** [comment]
@FloEdelmann great! :)

**Comment by RubenKelevra** [comment]
> Ah, that makes it clear, thanks. I'm not sure if it's okay for StreetComplete to start tagging that new tag though. There are already some (few) uses of the tag: https://taginfo.openstreetmap.org/keys/cycle_barrier#values
> 
> @westnordost What's your opinion on that?

So have we reached a decision? 🤔 😁

**Comment by westnordost** [comment]
Sorry I don't understand what this is about. What do cycle barriers to do with bollards?

**Comment by FloEdelmann** [comment]
@westnordost See https://github.com/streetcomplete/StreetComplete/issues/2128#issuecomment-849539675:

> Well, using the value of a key as subkey is pretty common. I doubt someone would complain if we 'invent' `cycle_barrier=removeable` as an analog to `bollard=removeable`.
> 
> This info is just very uncommon to be added right now, but this would follow established procedures and is analog to a very similar tag.

**Comment by westnordost** [comment]
Okay, but what have cycle barriers to do with bollards? This issue is about a quest for bollards, not for cycle barriers. `cycle_barrier=*` is barely used and not documented.
